packages/effect/src/internal/core-effect.ts

```diff
@@ -948,6 +948,20 @@ export const logAnnotations: Effect.Effect<HashMap.HashMap<string, unknown>> = c
     core.currentLogAnnotations
   )
 
+/** @internal */
+export const whenLogLevel = dual<
+  (level: LogLevel.LogLevel) => <A, E, R>(effect: Effect.Effect<A, E, R>) => Effect.Effect<Option.Option<A>, E, R>,
+  <A, E, R>(effect: Effect.Effect<A, E, R>, level: LogLevel.LogLevel) => Effect.Effect<Option.Option<A>, E, R>
+>(2, (effect, level) => {
+  return core.withFiberRuntime((fiberState) => {
+    const currentLogLevel = FiberRef.currentMinimumLogLevel.pipe(fiberState.getFiberRef)
```

Comment:
No need for the .pipe here

Reply:
fixed in [1dce660](https://github.com/Effect-TS/effect/pull/4342/commits/1dce6600254f45c64ae02f018aa704622dc416c4), I've also taken care of aligning the code used to filter with the one used in `FiberRuntime.log` just to keep it consistent (e.g. moving from `LogLevel.lessThanEqual` to `LogLevel.greaterThan` and doing an early return.